en/computer_vision/path_planning_interface.md

```diff
@@ -0,0 +1,99 @@
+# Path Planning Offboard Interface
+
+PX4 uses a number of MAVLink interfaces for integrating path planning services from a companion computer (including obstacle avoidance in missions, [safe landing](../computer_vision/safe_landing.md), and future services):
+- The [MAVLink Path Planning Protocol](https://mavlink.io/en/services/trajectory.html) message [TRAJECTORY_REPRESENTATION_WAYPOINTS](https://mavlink.io/en/messages/common.html#TRAJECTORY_REPRESENTATION_WAYPOINTS) is used to sent the current and next waypoint, and receive a stream of setpoints for the new path.
+- The [HEARTBEAT/Connection Protool](https://mavlink.io/en/services/heartbeat.html) is used for "proof of life" detection.
+- [LOCAL_POSITION_NED](https://mavlink.io/en/messages/common.html#LOCAL_POSITION_NED) and [ALTITUDE](https://mavlink.io/en/messages/common.html#ALTITUDE) send the vehicle local position and altitude, respectively.
+
+> **Tip** The message flows from PX4 UORB topics, through MAVLink, to ROS and back again are all documented in: [PX4/avoidance > Message Flows](https://github.com/PX4/avoidance#message-flows).
+
+
+All services that use this interface send and receive messages of the same type/format.
+Developers can therefore use this interface to create their own new companion-side path planning services, or tweak the existing planner software.
+
+
+## PX4 Configuration
+
+Path planning is activated in PX4 by [setting](../advanced_config/parameters.md) the [COM_OBS_AVOID](../advanced_config/parameter_reference.md#COM_OBS_AVOID) to 1.
+
+## Companion Computer Setup
+
+Companion-side hardware setup and hardware/software configuration is provided in the [PX4/avoidance](https://github.com/PX4/avoidance#obstacle-detection-and-avoidance) Github repo.
+
+The actual setup/configuration required depends on the planner being used.
+
+> **Warning** Only one planner can run on the companion computer at a time (at time of writing).
```

Comment:
Is this correct?ie you can only run one planner, and therefore get whatever features it offers.

Is there a plan to work out how to support multiple planners or some other approach to support all the features at the same time? It is a bit of a marketing fail to say we support obstacle avoidance or safe landing as opposed to saying we support "and".

Reply:
At the moment it is 'or', as the safe landing planner is brand new. We'll work on getting them running together, but it isn't top priority.